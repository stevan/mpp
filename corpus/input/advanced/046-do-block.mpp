my $result = do {
    my $x = 10;
    my $y = 20;
    $x + $y;