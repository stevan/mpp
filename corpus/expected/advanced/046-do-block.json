[
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$result"
    },
    "initializer": {
      "type": "DoBlock",
      "statements": [
        {
          "type": "Declaration",
          "declarator": "my",
          "variable": {
            "type": "Variable",
            "name": "$x"
          },
          "initializer": {
            "type": "Number",
            "value": "10"
          }
        },
        {
          "type": "Declaration",
          "declarator": "my",
          "variable": {
            "type": "Variable",
            "name": "$y"
          },
          "initializer": {
            "type": "Number",
            "value": "20"
          }
        },
        {
          "type": "BinaryOp",
          "operator": "+",
          "left": {
            "type": "Variable",
            "name": "$x"
          },
          "right": {
            "type": "Variable",
            "name": "$y"
          }
        }
      ]
    }
  }
]